src/shared/condition.c

```diff
@@ -958,6 +960,118 @@ static int condition_test_file_is_executable(Condition *c, char **env) {
                 (st.st_mode & 0111));
 }
 
+static int condition_test_psi(Condition *c, char **env) {
+        _cleanup_free_ char *first = NULL, *second = NULL, *pressure_path = NULL;
+        loadavg_t *current, limit;
+        ResourcePressure pressure;
+        const char *p, *value;
+        OrderOperator order;
+        int r;
+
+        assert(c);
+        assert(c->parameter);
+        assert(IN_SET(c->type, CONDITION_MEMORY_PRESSURE, CONDITION_CPU_PRESSURE));
+
+        if (!is_pressure_supported())
+                return log_debug_errno(SYNTHETIC_ERRNO(EOPNOTSUPP), "Pressure Stall Information (PSI) is not supported");
+
+        p = c->parameter;
+        r = extract_many_words(&p, ":", 0, &first, &second, NULL);
+        if (r <= 0)
+                return log_debug_errno(r ?: SYNTHETIC_ERRNO(EINVAL), "Failed to parse condition parameter %s: %m", c->parameter);
+        /* If only one parameter is passed, then we look at the global system pressure rather than a specific cgroup. */
+        if (r == 1) {
+                pressure_path = path_join("/proc/pressure", c->type == CONDITION_CPU_PRESSURE ? "cpu" : "memory");
+                if (!pressure_path)
+                        return -ENOMEM;
+
+                value = first;
+        } else {
+                CGroupMask mask, required_mask = c->type == CONDITION_CPU_PRESSURE ? CGROUP_MASK_CPU : CGROUP_MASK_MEMORY;
+                const char *controller = c->type == CONDITION_CPU_PRESSURE ? "cpu.pressure" : "memory.pressure";
+                _cleanup_free_ char *slice_path = NULL;
+
+                r = cg_all_unified();
+                if (r < 0)
+                        return log_debug_errno(r, "Failed to determine whether the unified cgroups hierarchy is used: %m");
+                if (r == 0)
```

Comment:
here too: i think we should handle this gracefully and assume the system is not overloaded